pennylane/_qubit_device.py

```diff
@@ -279,7 +279,18 @@ def execute(self, circuit, **kwargs):
         # generate computational basis samples
         sample_type = (SampleMP, CountsMP, ClassicalShadowMP, ShadowExpvalMP)
         if self.shots is not None or any(isinstance(m, sample_type) for m in circuit.measurements):
+            # Lightning does not support apply(rotations) anymore, so we need to rotate here
+            if hasattr(self, "name") and isinstance(self.name, str) and "Lightning" in self.name:
+                diagonalizing_gates = self._get_diagonalizing_gates(circuit)
+                if diagonalizing_gates:
+                    self.apply_lightning(diagonalizing_gates)
             self._samples = self.generate_samples()
+            if hasattr(self, "name") and isinstance(self.name, str) and "Lightning" in self.name:
+                diagonalizing_gates = self._get_diagonalizing_gates(circuit)
+                if diagonalizing_gates:
+                    self.apply_lightning(
+                        [qml.adjoint(g, lazy=False) for g in reversed(diagonalizing_gates)]
+                    )
```

Comment:
Should we add tests (either in lightning or PL)?

Reply:
There are tests in Lightning. If I add tests here, we'd have to change the CI to target Lightning's RC branch. Is that worth the trouble? Or can we add tests after releasing v0.35.1 in Lightning?

Reply:
[link to tests](https://github.com/PennyLaneAI/pennylane-lightning/pull/633/files#diff-37b15c763c59209952b8d4990702e5acb4455ba667cf54c8e4c6967f7e155b8aR683)

Reply:
I think it depends. Would v0.35.1 of PennyLane work with v0.35.0 of Lightning after this change?

If yes, then it could be worth adding a test here. If not, then we might be OK with leaving this as-is.

Additionally, if the answer is "no", we may need to update this line of `setup.py`
```python
    "pennylane-lightning>=0.35",
```
to
```python
    "pennylane-lightning>=0.35.1",
```

Reply:
Good point. Most features supported by Lightning should work, but sampling is currently broken. We could probably write a test that passes and cover these lines, but to make this work we need the corresponding fix in Lightning. We usually release Lightning before PennyLane and then have PennyLane depend on the most recent Lightning version. I think the best course of action would be to follow that to get the most out of this bug fix. So yes, I think we might need the change `    "pennylane-lightning>=0.35.1",`.